### PR TITLE
Fix issues with popup in live

### PR DESCRIPTION
**Pull request by rikterbeek**
**Description**

**Tested scenarios**

**Fixed issue**:

### CODE REVIEW

File: view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js [hunk 1]
```diff
@@ -246,24 +246,40 @@ define(
...
                     self.threeDS2ChallengeComponent.mount(threeDS2Node);
                 }
             },
+            /**
+             * This method is a workaround to close the modal in the right way and reconstruct the threeDS2Modal.
+             * This will solve issues when you cancel the 3DS2 challenge and retry the payment
+             */
+            closeModal: function (popupModal) {
+                popupModal.modal("closeModal");
+                $('.threeDS2Modal').remove();
```
Comment (cyattilakiss) [inline view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js]: Isn't it an id?
```suggestion
                $('#threeDS2Modal').remove();
```

File: view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js [hunk 2]
```diff
@@ -246,24 +246,40 @@ define(
...
                             onError: function (error) {
+                                self.closeModal(popupModal);
                                 console.log(JSON.stringify(error));
                             }
                         });
                     self.threeDS2ChallengeComponent.mount(threeDS2Node);
                 }
             },
+            /**
+             * This method is a workaround to close the modal in the right way and reconstruct the threeDS2Modal.
```
Comment (cyattilakiss) [inline view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js]: There supposed to be a function for the checkout component which destroys the component itself and makes it available to mount other components to the div in the DOM. I like that approach more than removing then adding the elements with JS to the DOM but I would say let's merge it and then I will give it a look later.

File: view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js (hunk 1 above)
Comment (rikterbeek) [reply view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js]: no it creates an class for this by the modal

File: view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js (hunk 2 above)
Comment (rikterbeek) [reply view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js]: This is not related to this. But the only workaround we found to destroy the modal completly.

**Pull request merged**